model/atmosphere/dycore/src/icon4py/model/atmosphere/dycore/stencils/compute_horizontal_velocity_quantities.py

```diff
@@ -0,0 +1,269 @@
+    contravariant_correction_at_edges_on_model_levels: fa.EdgeKField[ta.vpfloat],
+    vn: fa.EdgeKField[ta.wpfloat],
+    e_flx_avg: gtx.Field[gtx.Dims[dims.EdgeDim, dims.E2C2EODim], ta.wpfloat],
+    geofac_grdiv: gtx.Field[gtx.Dims[dims.EdgeDim, dims.E2C2EODim], ta.wpfloat],
+    rbf_vec_coeff_e: gtx.Field[gtx.Dims[dims.EdgeDim, dims.E2C2EDim], ta.wpfloat],
+    rho_at_edges_on_model_levels: fa.EdgeKField[ta.wpfloat],
+    theta_v_at_edges_on_model_levels: fa.EdgeKField[ta.wpfloat],
+    ddqz_z_full_e: fa.EdgeKField[ta.vpfloat],
+    ddxn_z_full: fa.EdgeKField[ta.vpfloat],
+    ddxt_z_full: fa.EdgeKField[ta.vpfloat],
+    wgtfac_e: fa.EdgeKField[ta.vpfloat],
+    wgtfacq_e: fa.EdgeKField[ta.vpfloat],
+    nflatlev: gtx.int32,
+    horizontal_start: gtx.int32,
+    horizontal_end: gtx.int32,
+    vertical_start: gtx.int32,
+    vertical_end: gtx.int32,
+):
+    _compute_horizontal_velocity_quantities_and_fluxes(
+        contravariant_correction_at_edges_on_model_levels,
+        vn,
+        e_flx_avg,
+        geofac_grdiv,
+        rbf_vec_coeff_e,
+        rho_at_edges_on_model_levels,
+        theta_v_at_edges_on_model_levels,
+        ddqz_z_full_e,
+        ddxn_z_full,
+        ddxt_z_full,
+        wgtfac_e,
+        nflatlev,
+        out=(
+            spatially_averaged_vn,
+            horizontal_gradient_of_normal_wind_divergence,
+            tangential_wind,
+            mass_flux_at_edges_on_model_levels,
+            theta_v_flux_at_edges_on_model_levels,
+            vn_on_half_levels,
+            tangential_wind_on_half_levels,
+            horizontal_kinetic_energy_at_edges_on_model_levels,
+            contravariant_correction_at_edges_on_model_levels,
+        ),
+        domain={
+            dims.EdgeDim: (horizontal_start, horizontal_end),
+            dims.KDim: (vertical_start, vertical_end - 1),
+        },
+    )
+
+    _extrapolate_at_top(
+        wgtfacq_e,
+        vn,
+        out=vn_on_half_levels,
+        domain={
+            dims.EdgeDim: (horizontal_start, horizontal_end),
+            dims.KDim: (vertical_end - 1, vertical_end),
+        },
+    )
+
+
+@gtx.field_operator
+def _compute_averaged_vn_and_fluxes_and_prepare_tracer_advection(
+    substep_and_spatially_averaged_vn: fa.EdgeKField[ta.wpfloat],
+    substep_averaged_mass_flux: fa.EdgeKField[ta.wpfloat],
+    e_flx_avg: gtx.Field[gtx.Dims[dims.EdgeDim, dims.E2C2EODim], ta.wpfloat],
+    vn: fa.EdgeKField[ta.wpfloat],
+    rho_at_edges_on_model_levels: fa.EdgeKField[ta.wpfloat],
+    ddqz_z_full_e: fa.EdgeKField[ta.vpfloat],
+    theta_v_at_edges_on_model_levels: fa.EdgeKField[ta.wpfloat],
+    prepare_advection: bool,
+    at_first_substep: bool,
+    r_nsubsteps: ta.wpfloat,
+) -> tuple[
+    fa.EdgeKField[ta.wpfloat],
+    fa.EdgeKField[ta.wpfloat],
+    fa.EdgeKField[ta.wpfloat],
+    fa.EdgeKField[ta.wpfloat],
+    fa.EdgeKField[ta.wpfloat],
+]:
+    spatially_averaged_vn = _compute_avg_vn(e_flx_avg, vn)
```

Comment:
should it be called 
```suggestion
    spatially_averaged_vn = _compute_spatial_edge_average(e_flx_avg, vn)
```
in an effort to make operators more general?

Reply:
Ideally we would have a feature in gt4py to bind `e_flx_avg` to that function, then I would like the flow:
`_compute_spatial_edge_average(vn)`.